service/history/api/create_workflow_util.go

```diff
@@ -170,9 +170,14 @@ func NewWorkflowWithSignal(
 
 // NOTE: must implement CreateLeaseFunc.
 func NewWorkflowLeaseAndContext(
+	existingLease WorkflowLease,
 	shardCtx shard.Context,
 	ms workflow.MutableState,
 ) (WorkflowLease, error) {
+	// TODO(stephanos): remove this hack
+	if existingLease != nil {
+		return existingLease, nil
+	}
```

Comment:
This makes sure this is a noop when called with an existing lease. Which only happens in exactly one case (see below).